Formula/petsc-complex.rb

```diff
@@ -1,10 +1,14 @@
 class PetscComplex < Formula
   desc "Portable, Extensible Toolkit for Scientific Computation (complex)"
   homepage "https://www.mcs.anl.gov/petsc/"
-  url "https://ftp.mcs.anl.gov/pub/petsc/release-snapshots/petsc-lite-3.14.4.tar.gz"
-  sha256 "b030969816e02c251a6d010c07a90b69ade44932f9ddfac3090ff5e95ab97d5c"
+  url "https://ftp.mcs.anl.gov/pub/petsc/release-snapshots/petsc-lite-3.14.5.tar.gz"
+  sha256 "8b8ff5c4e10468f696803b354a502d690c7d25c19d694a7e10008a302fdbb048"
   license "BSD-2-Clause"
-  revision 1
+
+  livecheck do
+    url "https://www.mcs.anl.gov/petsc/download/index.html"
```

Comment:
```suggestion
    url "https://www.mcs.anl.gov/petsc/download/"
```

Omitting the trailing `index.html` works and is perhaps a little tidier. If you're good with this, I'll update the URL in `petsc` in a follow-up PR to match.